app/models/merchant.rb

```diff
@@ -1,6 +1,12 @@
 class Merchant < ApplicationRecord
   has_many :items
   has_many :invoices
+  has_many :invoice_items, through: :items
   has_many :transactions, through: :invoices
   has_many :customers, through: :invoices
+
+  def self.most_items_sold(quantity)
+    Merchant.select("merchants.*, SUM(invoice_items.quantity) AS most_items").
+      joins(:invoice_items).order("most_items DESC").group("merchants.id").first(quantity)
```

Comment:
@pwentz Since the quantity here will be coming in from `params` in the controller, it may be more secure to handle the argument for `.first` as `.first("?", quantity)`. This should protect against SQL injection.